kinematic_properties.py

```diff
@@ -242,24 +240,77 @@ def get_inertia_tensor(
        Masses of the particles.
      - position: unyt.unyt_array
        Positions of the particles.
+     - radius: unyt.unyt_quantity
+       Exclude particles outside this radius for the inertia tensor calculation
+     - reduced: bool
+       Whether to calculate the reduced inertia tensor
+     - max_iterations: int
+       The maximum number of iterations to repeat the inertia tensor calculation
 
     Returns the inertia tensor.
-    """
-    # 3x3 inertia tensor
-    Itensor = (
-        mass[:, None, None] / mass.sum() * position[:, None:, None] * position[:, None]
-    ).sum(axis=0)
-
-    # Symmetric, so only return lower triangle
-    Itensor = np.concatenate([np.diag(Itensor), Itensor[np.triu_indices(3, 1)]])
-
-    return Itensor
-
-
-def get_projected_inertia_tensor(
-    mass: unyt.unyt_array, position: unyt.unyt_array, axis: int
-) -> unyt.unyt_array:
-    """
+    '''
+    # Shouldn't calculate an inertia tensor for small numbers of particles
+    if position.shape[0] < 20:
+        return None
+
+    # Remove particles at centre if calculating reduced tensor
```

Comment:
Why is this needed? Will it blow up the value otherwise?

Reply:
Yes https://github.com/SWIFTSIM/SOAP/blob/5107c1f26880497e774c890bb0e06edaf4838b52/kinematic_properties.py#L398 won't be happy